File: opspilot-enterprise/packages/shared-types/src/agent-run.ts
export type AgentRunStatus = "queued" | "running" | "completed" | "failed" | "cancelled";
export type AgentStepStatus = "waiting" | "running" | "done" | "failed" | "skipped";

export interface AgentRunStep {
  step_id: string;
  agent_name: string;
  status: AgentStepStatus;
  input_summary: string;
  output_summary: string | null;
  tool_calls: number;
  started_at: string | null;
  completed_at: string | null;
  duration_ms: number | null;
  error: string | null;
}

export interface AgentRun {
  id: string;
  intent: string;
  status: AgentRunStatus;
  trigger: "user" | "schedule" | "alert" | "api";
  incident_ref: string | null;
  session_ref: string | null;
  steps: AgentRunStep[];
  total_tool_calls: number;
  total_duration_ms: number | null;
  started_at: string;
  completed_at: string | null;
  output_summary: string | null;
  error: string | null;
}

File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/agent_run.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel


class AgentRunStep(BaseModel):
    step_id: str
    agent_name: str
    status: str
    input_summary: str
    output_summary: Optional[str] = None
    tool_calls: int = 0
    started_at: Optional[str] = None
    completed_at: Optional[str] = None
    duration_ms: Optional[int] = None
    error: Optional[str] = None


class AgentRun(BaseModel):
    id: str
    intent: str
    status: str
    trigger: str
    incident_ref: Optional[str] = None
    session_ref: Optional[str] = None
    steps: list[AgentRunStep] = []
    total_tool_calls: int = 0
    total_duration_ms: Optional[int] = None
    started_at: str
    completed_at: Optional[str] = None
    output_summary: Optional[str] = None
    error: Optional[str] = None
